### PR TITLE
Support MBR on S905x

### PR DESCRIPTION
**Pull request by carlocaione**
https://phabricator.endlessm.com/T10685

### CODE REVIEW

File: eos-extra-resize [hunk 1]
```diff
@@ -43,10 +50,13 @@ if [ "$marker" = "GUID:55" ]; then
...
     part_start=$(< /sys/class/block/$extra_part_name/start)
 
-    # The partition table is recreated below, which also includes the
-    # magic 55 attribute being dropped.
+    # The partition table is recreated below
     echo "Try to resize $extra_part at sector $part_start to fill $extra_disk"
-    echo "$part_start" | sfdisk --force --no-reread --label gpt $extra_disk
+    echo "$part_start" | sfdisk --force --no-reread --label $pt_label $extra_disk
+
+    # For MBR the marker is removed in endless-repartition.sh when rewriting
```
Comment (dsd) [inline eos-extra-resize]: Actually for the extra disk, for both GPT and MBR the marker is removed in this script 2 lines above, we rewrite the whole partition table in that sfdisk command.

File: eos-extra-resize [hunk 2]
```diff
@@ -32,9 +32,16 @@ if [ -z "$extra_disk" ]; then
     exit 0
 fi
 
+pt_label=$(blkid -o value -s PTTYPE $extra_disk)
+
+if [ "$pt_label" = "gpt" ]; then
+  marker=$(sfdisk --force --part-attrs $extra_disk 1)
+else
+  marker=$(sfdisk --force --part-type $extra_disk 4)
```
Comment (dbnicholson) [inline eos-extra-resize]: Am I wrong, or are you reading the wrong partition in the MBR case? There aren't 4 partitions in the extra disk.

File: dracut/repartition/endless-repartition.sh [hunk 3]
```diff
@@ -80,9 +80,18 @@ if [ -z "${root_disk}" ]; then
   exit 0
 fi
 
+pt_label=$(blkid -o value -s PTTYPE $root_disk)
+
 # Check for our magic "this is Endless" marker
-marker=$(sfdisk --force --part-attrs $root_disk $partno)
-if [ "$marker" != "GUID:55" ]; then
+if [ "$pt_label" = "dos" ]; then
+  marker=$(sfdisk --force --part-type $root_disk 4)
```
Comment (dbnicholson) [inline dracut/repartition/endless-repartition.sh]: Any reason you're using `4` here instead of `$partno`?

Comment (carlocaione) [inline dracut/repartition/endless-repartition.sh]: Uhm, `4` is wrong but also `$partno` is not correct either. `$partno` is the partition number for the rootfs where is the marker for the GPT case. For the DOS case we have a new additional partition so it should be `$partno+1` if I'm not mistaken.

Comment (dsd) [inline dracut/repartition/endless-repartition.sh]: In the previous implementation we always put the marker on partition 4 (which of course is not a real partition).

Comment (carlocaione) [inline dracut/repartition/endless-repartition.sh]: Right. Working on this after a while made me forget these details. 
We are _still_ using the partition `4` for this, so this line is actually correct. Thank for reminding this.

File: eos-extra-resize (hunk 2 above)
Comment (carlocaione) [inline eos-extra-resize]: As before we are using `printf "\xdd" | dd of=${disk1_img} bs=1 count=1 seek=498 conv=notrunc` in `eos-image-builder` to write a fake partition `4`

File: eos-extra-resize (hunk 1 above)
Comment (carlocaione) [inline eos-extra-resize]: yeah, right.

File: eos-extra-resize (hunk 2 above)
Comment (dbnicholson) [inline eos-extra-resize]: OK.

**Pull request merged**